core/lib/makeswift/components/site-theme/client.tsx

```diff
@@ -0,0 +1,121 @@
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+'use client';
+
+type Font = {
+  fontFamily: string;
+};
+
+// eslint-disable-next-line @typescript-eslint/consistent-indexed-object-style
+type ThemeProps = {
+  [key in string]?: string | number | Font | ThemeProps;
+};
+
+function isFont(value: unknown): value is Font {
+  if (typeof value !== 'object' || value === null) {
+    return false;
+  }
+
+  const keys = Object.keys(value);
+
+  return keys.length === 1 && 'fontFamily' in value && typeof value.fontFamily === 'string';
+}
```

Comment:
Would it be worth it to use Zod here instead of writing custom parsing logic?

Reply:
I considered it; I didn't think it was worth it since this check is trivial, our theme is quite large, and this is called during render on every theme prop that's not a string or a number.